framework/src/base/Assembly.C

```diff
@@ -1269,6 +1272,11 @@ Assembly::reinitFEFace(const Elem * elem, unsigned int side)
 
   if (_xfem != nullptr)
     modifyFaceWeightsDueToXFEM(elem, side);
+
+  unsigned int n = _extra_elem_ids.size() - 1;
+  for (unsigned int i = 0; i < n; ++i)
+    _extra_elem_ids[i] = _current_elem->get_extra_integer(i);
+  _extra_elem_ids[n] = _current_elem->subdomain_id();
```

Comment:
Why do you copy the subdomain id into this `_extra_elem_ids` vector?

Reply:
Because we want to use the same postprocessing objects for handling either subdomain ID or something ID like material ID. With subdomain ID in this, we can simply switch the operation by using a different element ID index.

Reply:
But keep in mind that you are adding this copy operation to every element and face reinit, whether extra IDs are used or not. I'm not convinced that a minor convenience in postprocessing is worth that extra overhead.

Reply:
The only extra cost is one integer copy though. I initially did not have this and realized that my objects are super ugly. Another argument is that we copied the subdomain IDs anyway in `Assembly`. With the copying here, we can potentially remove the other copying. I do not want to introduce extra operations in the critical codes as well. But here I have to do this considering the significant simplifications in down-stream objects.